test/stat_tracker_test.rb

```diff
@@ -38,4 +38,73 @@ def test_it_stores_a_collection_of_game_teams_data
     assert_instance_of GameTeams, @stat_tracker.game_teams_data.sample
   end
 
+  def test_it_finds_highest_total_score
+    assert_equal 9, @stat_tracker.highest_total_score
+  end
+
+  def test_it_finds_lowest_total_score
+    assert_equal 0, @stat_tracker.lowest_total_score
+  end
+
+  def test_it_finds_biggest_blowout
+    assert_equal 5, @stat_tracker.biggest_blowout
+  end
+
+  def test_it_finds_percentage_home_wins
+    assert_equal 38.43, @stat_tracker.percentage_home_wins
+  end
+
+  def test_it_finds_percentage_visitor_wins
+    assert_equal 39.49, @stat_tracker.percentage_visitor_wins
+  end
+
+  def test_it_finds_percentage_of_ties
+    assert_equal 22.08, @stat_tracker.percentage_ties
+  end
+
+  def test_it_counts_games_by_season
+    expected = {
+      '20122013' => 51,
+      '20132014' => 82,
+      '20142015' => 85,
+      '20152016' => 87,
+      '20162017' => 82,
+      '20172018' => 84
+    }
+
+    assert_equal expected, @stat_tracker.count_of_games_by_season
+  end
+
+  def test_it_calculates_average_goals_per_game
+    assert_equal 4.27, @stat_tracker.average_goals_per_game
+  end
+
+  def test_it_calculates_total_goals_per_season
```

Comment:
This method isn't called for in the Interaction pattern. I split it out as a helper method for average goals per season to keep overall method length at an acceptable level.